src/node_sets.cpp

```diff
@@ -73,6 +77,21 @@ class NodeSets
         parse_compound(j, node_sets_);
     }
 
+    explicit NodeSets(const fs::path& path)
+        : NodeSets(json::parse(std::fstream(path))) {}
+
+    explicit NodeSets(const std::string& content)
+        : NodeSets(json::parse(content)) {}
```

Comment:
This will allow `detail::NodeSets("nodesets.h5")` to compile, it won't parse (because of missing inner `"`).

Reply:
I'm not sure I follow this point, could you expand?  Is there a test I should add that would catch what you're suggesting?

Reply:
I tend to not distinguish between path object and strings. Often a string is a perfectly fine way of specifying a path. Hence when I read the first ctor, I think: "Ah, it accepts paths." Good, let's pass it a `"nodeset.h5"`, resulting in something like:
```
auto detail_nodeset = detail::NodeSet("foo.h5");
```
thinking that this will be the same as `detail::NodeSet(fs::path("foo.h5"))`. It's not. The slightly unfortunate thing is that it'll compile. The good news is that it'll fail, because `"foo.h5"` is not a valid JSON object `"\"foo.h5\""` would be.

I understand it's all happening inside `detail` so it's not user facing. Hence there's likely little cause to have a path which is an `std::string`.

Is this a bit more clear? (There's nothing deep or serious happening here.)

Reply:
Ah, thank you, yes, that's perfectly clear now.

I also see strings as paths (bad habits die hard), but since it's internal, I'm going to leave it as it is.

The best plan would have been to not have any ctors, and just a `fromFile`, `fromContents`, to avoid this ambiguity.